Mathlib/CategoryTheory/Enriched/Basic.lean

```diff
@@ -94,6 +94,70 @@ theorem e_assoc' (W X Y Z : C) :
       eComp V W X Y ▷ _ ≫ eComp V W Y Z := by
   rw [← e_assoc V W X Y Z, Iso.hom_inv_id_assoc]
 
+/-- Isomorphisms in a `V`-enriched category `C`consist of a morphism
+`𝟙_ V ⟶ X ⟶[V] Y`, an inverse `𝟙_ V ⟶ Y ⟶[V] X`, and proofs that these compose to the identity
+morphism. -/
+@[ext]
+structure EnrichedIso (X Y : C) where
+  /-- The forward direction of an isomorphism. -/
+  hom : 𝟙_ V ⟶ X ⟶[V] Y
+  /-- The backward direction of an isomorphism. -/
+  inv : 𝟙_ V ⟶ Y ⟶[V] X
+  hom_inv : (λ_ _).inv ≫ (hom ⊗ₘ inv) ≫ eComp V X Y X = eId V X := by aesop_cat
+  inv_hom : (λ_ _).inv ≫ (inv ⊗ₘ hom) ≫ eComp V Y X Y = eId V Y := by aesop_cat
+
+namespace EnrichedIso
+
+variable {V}
+
+/-- The identity isomorphism in a `V`-enriched category. -/
+@[refl, simps]
+def refl (X : C) : EnrichedIso V X X where
+  hom := eId V X
+  inv := eId V X
+  hom_inv := by simp [tensorHom_def']
+  inv_hom := by simp [tensorHom_def']
+
+/-- The inverse isomorphism of an isomorphism in a `V`-enriched category. -/
+@[symm, simps]
+def symm {X Y : C} (I : EnrichedIso V X Y) : EnrichedIso V Y X where
+  hom := I.inv
+  inv := I.hom
+  hom_inv := I.inv_hom
+  inv_hom := I.hom_inv
+
+open EnrichedCategory
+
+/-- The composition of to isomorphisms in a `V`-enriched category. -/
+@[trans, simps]
+def trans {X Y Z : C} (I : EnrichedIso V X Y) (J : EnrichedIso V Y Z) : EnrichedIso V X Z where
+  hom := (λ_ _).inv ≫ (I.hom ⊗ₘ J.hom) ≫ eComp V X Y Z
+  inv := (λ_ _).inv ≫ (J.inv ⊗ₘ I.inv) ≫ eComp V Z Y X
+  hom_inv := by
```

Comment:
Just glancing at the code symmetry in `hom_inv` and `inv_hom` it looks like a lemma or two should be factored out of this.

Reply:
I guess the length is also a giveaway...